// react-notion-x/build/esm/utils.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import isUrl from 'is-url-superb';
import { format } from 'date-fns';
export var cs = function () {
    var classes = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        classes[_i] = arguments[_i];
    }
    return classes.filter(function (a) { return !!a; }).join(' ');
};
export { isUrl };
var groupBlockContent = function (blockMap) {
    var output = [];
    var lastType = undefined;
    var index = -1;
    Object.keys(blockMap).forEach(function (id) {
        var _a, _b;
        var blockValue = (_a = blockMap[id]) === null || _a === void 0 ? void 0 : _a.value;
        if (blockValue) {
            (_b = blockValue.content) === null || _b === void 0 ? void 0 : _b.forEach(function (blockId) {
                var _a, _b;
                var blockType = (_b = (_a = blockMap[blockId]) === null || _a === void 0 ? void 0 : _a.value) === null || _b === void 0 ? void 0 : _b.type;
                if (blockType && blockType !== lastType) {
                    index++;
                    lastType = blockType;
                    output[index] = [];
                }
                if (index > -1) {
                    output[index].push(blockId);
                }
            });
        }
        lastType = undefined;
    });
    return output;
};
export var getListNumber = function (blockId, blockMap) {
    var groups = groupBlockContent(blockMap);
    var group = groups.find(function (g) { return g.includes(blockId); });
    if (!group) {
        return;
    }
    return group.indexOf(blockId) + 1;
};
export var defaultMapImageUrl = function (url, block) {
    if (!url) {
        return null;
    }
    if (url.startsWith('data:')) {
        return url;
    }
    if (url.startsWith('/images')) {
        url = "https://www.notion.so" + url;
    }
    // more recent versions of notion don't proxy unsplash images
    if (!url.startsWith('https://images.unsplash.com')) {
        url = "https://www.notion.so" + (url.startsWith('/image') ? url : "/image/" + encodeURIComponent(url));
        var notionImageUrlV2 = new URL(url);
        var table = block.parent_table === 'space' ? 'block' : block.parent_table;
        if (table === 'collection') {
            table = 'block';
        }
        notionImageUrlV2.searchParams.set('table', table);
        notionImageUrlV2.searchParams.set('id', block.id);
        notionImageUrlV2.searchParams.set('cache', 'v2');
        url = notionImageUrlV2.toString();
    }
    return url;
};
export var defaultMapPageUrl = function (rootPageId) { return function (pageId) {
    pageId = (pageId || '').replace(/-/g, '');
    if (rootPageId && pageId === rootPageId) {
        return '/';
    }
    else {
        return "/" + pageId;
    }
}; };
export var getHashFragmentValue = function (url) {
    return url.includes('#') ? url.replace(/^.+(#.+)$/, '$1') : '';
};
var months = [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec'
];
export var formatDate = function (input) {
    var date = new Date(input);
    var month = date.getMonth();
    return months[month] + " " + date.getUTCDate() + ", " + date.getUTCFullYear();
};
export var isBrowser = typeof window !== 'undefined';
export function getCollectionGroups(collection, collectionView, collectionData) {
    var _a;
    var rest = [];
    for (var _i = 3; _i < arguments.length; _i++) {
        rest[_i - 3] = arguments[_i];
    }
    var elems = ((_a = collectionView === null || collectionView === void 0 ? void 0 : collectionView.format) === null || _a === void 0 ? void 0 : _a.collection_groups) || [];
    return elems === null || elems === void 0 ? void 0 : elems.map(function (_a) {
        var _b, _c;
        var property = _a.property, hidden = _a.hidden, _d = _a.value, value = _d.value, type = _d.type;
        var isUncategorizedValue = typeof value === 'undefined';
        var isDateValue = value === null || value === void 0 ? void 0 : value.range;
        // TODO: review dates reducers
        var queryLabel = isUncategorizedValue
            ? 'uncategorized'
            : isDateValue
                ? ((_b = value.range) === null || _b === void 0 ? void 0 : _b.start_date) || ((_c = value.range) === null || _c === void 0 ? void 0 : _c.end_date)
                : (value === null || value === void 0 ? void 0 : value.value) || value;
        var collectionGroup = collectionData["results:" + type + ":" + queryLabel];
        var queryValue = !isUncategorizedValue && (isDateValue || (value === null || value === void 0 ? void 0 : value.value) || value);
        var schema = collection.schema[property];
        // Checkbox boolen value must be Yes||No
        if (type === 'checkbox' && value) {
            queryValue = 'Yes';
        }
        if (isDateValue) {
            schema = {
                type: 'text',
                name: 'text'
            };
            // TODO: review dates format based on value.type ('week'|'month'|'year')
            queryValue = format(new Date(queryLabel), 'MMM d, YYY hh:mm aa');
        }
        return __assign({ collectionGroup: collectionGroup, schema: schema, value: queryValue || 'No description', hidden: hidden, collection: collection, collectionView: collectionView, collectionData: collectionData, blockIds: collectionGroup === null || collectionGroup === void 0 ? void 0 : collectionGroup.blockIds }, rest);
    });
}
