// notion-utils/build/esm/get-page-breadcrumbs.js
import { getBlockTitle } from './get-block-title';
import { getBlockIcon } from './get-block-icon';
import { getBlockParentPage } from './get-block-parent-page';
export var getPageBreadcrumbs = function (recordMap, activePageId) {
    var _a;
    var blockMap = recordMap.block;
    var breadcrumbs = [];
    var currentPageId = activePageId;
    do {
        var block = (_a = blockMap[currentPageId]) === null || _a === void 0 ? void 0 : _a.value;
        if (!block) {
            break;
        }
        var title = getBlockTitle(block, recordMap);
        var icon = getBlockIcon(block, recordMap);
        if (!(title || icon)) {
            break;
        }
        breadcrumbs.push({
            block: block,
            active: currentPageId === activePageId,
            pageId: currentPageId,
            title: title,
            icon: icon
        });
        var parentBlock = getBlockParentPage(block, recordMap);
        var parentId = parentBlock === null || parentBlock === void 0 ? void 0 : parentBlock.id;
        if (!parentId) {
            break;
        }
        currentPageId = parentId;
        // eslint-disable-next-line no-constant-condition
    } while (true);
    breadcrumbs.reverse();
    return breadcrumbs;
};
