// notion-utils/build/cjs/get-page-breadcrumbs.js
"use strict";
exports.__esModule = true;
exports.getPageBreadcrumbs = void 0;
var get_block_title_1 = require("./get-block-title");
var get_block_icon_1 = require("./get-block-icon");
var get_block_parent_page_1 = require("./get-block-parent-page");
var getPageBreadcrumbs = function (recordMap, activePageId) {
    var _a;
    var blockMap = recordMap.block;
    var breadcrumbs = [];
    var currentPageId = activePageId;
    do {
        var block = (_a = blockMap[currentPageId]) === null || _a === void 0 ? void 0 : _a.value;
        if (!block) {
            break;
        }
        var title = (0, get_block_title_1.getBlockTitle)(block, recordMap);
        var icon = (0, get_block_icon_1.getBlockIcon)(block, recordMap);
        if (!(title || icon)) {
            break;
        }
        breadcrumbs.push({
            block: block,
            active: currentPageId === activePageId,
            pageId: currentPageId,
            title: title,
            icon: icon
        });
        var parentBlock = (0, get_block_parent_page_1.getBlockParentPage)(block, recordMap);
        var parentId = parentBlock === null || parentBlock === void 0 ? void 0 : parentBlock.id;
        if (!parentId) {
            break;
        }
        currentPageId = parentId;
        // eslint-disable-next-line no-constant-condition
    } while (true);
    breadcrumbs.reverse();
    return breadcrumbs;
};
exports.getPageBreadcrumbs = getPageBreadcrumbs;
